test_agent/rust/src/main.rs

```diff
@@ -44,13 +56,41 @@ fn connect_to_socket(addr: &str, port: u16) -> Result<TcpStream, Box<dyn std::er
     }
 }
 
-async fn connect_and_receive() -> Result<(), Box<dyn std::error::Error>> {
+async fn create_zenoh_u_transport() -> Box<dyn UTransport> {
+    let uauthority = UAuthority {
+        name: Some("MyAuthName".to_string()),
+        number: Some(Number::Id(vec![1, 2, 3, 4])),
+        ..Default::default()
+    };
+    let uentity = UEntity {
+        name: "default.entity".to_string(),
+        id: Some(u32::from(rand::random::<u16>())),
+        version_major: Some(1),
+        version_minor: None,
+        ..Default::default()
+    };
+    dbg!("zenoh transport created successfully");
+
+    Box::new(
+        UPClientZenoh::new(Config::default(), uauthority, uentity)
+            .await
+            .unwrap(),
+    )
+}
+
+async fn connect_and_receive(transport_name: &str) -> Result<(), Box<dyn std::error::Error>> {
     let test_agent = connect_to_socket(TEST_MANAGER_ADDR.0, TEST_MANAGER_ADDR.1)?;
     let ta_to_tm_socket = connect_to_socket(TEST_MANAGER_ADDR.0, TEST_MANAGER_ADDR.1)?;
     let foo_listener_socket_to_tm = connect_to_socket(TEST_MANAGER_ADDR.0, TEST_MANAGER_ADDR.1)?;
 
-    let u_transport = UTransportSocket::new()?;
-    debug!("Socket transport created successfully");
+    #[allow(clippy::single_match_else)]
```

Comment:
Typically if you have to silence clippy, it's a good idea to leave a little note as to why. Could you place something similar to what I wrote so the next person knows why we're avoiding clippy? :laughing:

Reply:
oh yeah, makes sense :) will do